c/parallel/test/test_segmented_reduce.cpp

```diff
@@ -0,0 +1,427 @@
+
+#include <cuda_runtime.h>
+
+#include <cstdint>
+#include <cstdlib>
+#include <numeric>
+
+#include "cccl/c/types.h"
+#include "test_util.h"
+#include <cccl/c/segmented_reduce.h>
+
+class SegmentedReduceProgram
```

Comment:
At the moment, does this save us any compile time? 

If not, can we use the same test pattern as in `reduce.cu`, `scan.cu` etc. for consistency? We can tackle the problem of compilation time in a future PR and apply the same solution to all test files in `c/parallel`.

Reply:
using .cpp over .cu definitely speeds things up, but one step at a time is fine